Riot/Modules/Application/AppCoordinator.swift

```diff
@@ -322,15 +322,39 @@ fileprivate class AppNavigator: AppNavigatorProtocol {
         self.appCoordinator.navigate(to: destination)
     }
     
-    func addLoadingActivity() -> Activity {
-        let presenter = ActivityIndicatorToastPresenter(
-            text: VectorL10n.roomParticipantsSecurityLoading,
-            navigationController: appNavigationVC
-        )
-        let request = ActivityRequest(
-            presenter: presenter,
-            dismissal: .manual
-        )
+    func addAppActivity(_ type: AppActivityType) -> Activity {
+        let request = activityRequest(for: type)
         return ActivityCenter.shared.add(request)
     }
+    
+    // MARK: - Private
+    
+    private func activityRequest(for type: AppActivityType) -> ActivityRequest {
+        switch type {
+        case let .loading(label):
+            let presenter = ToastActivityPresenter(
+                viewState: .init(
```

Comment:
It's a bit weird that the presenter is advertising that it's using a `RoundedToastView` by requiring one of its viewStates. That's an implementation detail.

Reply:
Indeed, I extracted the view state out of the context of `RoundedToastView` so now it can be a shared type between the presenter and the view